pkg/api/http/server.go

```diff
@@ -8,24 +8,55 @@ import (
 	"github.com/cloudwego/hertz/pkg/common/utils"
 	"github.com/cloudwego/hertz/pkg/protocol/consts"
 	"time"
+	"glide/pkg/api"
+	"sync"
+	"log/slog"
+	
 )
 
 type HTTPServer struct {
 	server *server.Hertz
 }
 
+type ServerManager struct {
+	httpServer *HTTPServer
+	shutdownWG *sync.WaitGroup
+}
+
 func NewHttpServer(config *HTTPServerConfig) (*HTTPServer, error) {
 	return &HTTPServer{
 		server: config.ToServer(),
 	}, nil
 }
 
+// This func has the routes for the server
 func (srv *HTTPServer) Run() error {
-	srv.server.GET("/health", func(ctx context.Context, c *app.RequestContext) {
-		c.JSON(consts.StatusOK, utils.H{"healthy": true})
-	})
+    srv.server.GET("/health", func(ctx context.Context, c *app.RequestContext) {
+        c.JSON(consts.StatusOK, utils.H{"healthy": true})
+    })
+
+    srv.server.POST("/chat", func(ctx context.Context, c *app.RequestContext) {
+
+		slog.Info("post request at /chat received")
+
+        // Pass the client request body to SendRequest
+        resp, err := api.Router(c, "/chat")
+		
+
+		slog.Info("provider response received")
```

Comment:
Did you run `make lint` on this PR? Until we have CI in place, we need to keep that linted manually, unfortunately.